### DISCUSSION TITLE
Should we release 0.71.14?

### DISCUSSION THREAD

**Discussion by fortmarek** [reactwg/react-native-releases#87] [Patches]
## Released, please refer to #90

---

Current Release: [0.71.13](https://github.com/facebook/react-native/releases/tag/v0.71.13)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of qualified picks

1. https://github.com/facebook/react-native/commit/f6107b6c79c03182dcbe58ea7ba84c12cb77f756
2. https://github.com/facebook/react-native/commit/5a671c130bbd25a4d6c571884b9d5c800dce07e9
3. https://github.com/facebook/react-native/commit/647e09dc4c414f4889af01ecdcd39e89a30f1ff7
4. https://github.com/facebook/react-native/commit/462d93d9c18b453f0c14c15cf44c8d1de280edb9
5. https://github.com/facebook/react-native/commit/8b1bf058c4bcbf4e5ca45b0056217266a1ed870c
6. https://github.com/facebook/react-native/commit/3eeee11d7ac4075d0917233d3be4a9469f802d35
7. https://github.com/facebook/react-native/commit/ce39931bc2b02f13cbc5751ba4d4a6dbc07bc91a

#### Local commits to backport to main

1.

**Comment by fortmarek** [comment]
Left-over pick from the 0.71.13 discussion: https://github.com/reactwg/react-native-releases/discussions/78#discussioncomment-6764208

**Reply by fortmarek** [reply]
Picked ✅

**Comment by erinleigh90** [comment]
Can we backport [facebook/react-native@462d93d](https://github.com/facebook/react-native/commit/462d93d9c18b453f0c14c15cf44c8d1de280edb9) to 0.71.14? (Removes the unset-value@1.0.0 dependency which has been flagged as vulnerable)

**Reply by fortmarek** [reply]
Picked ✅

**Comment by ken0nek** [comment]
Hi,

I would like to have this change for 0.71.14 (and also other version patches).

https://github.com/facebook/react-native/pull/39060

**Reply by fortmarek** [reply]
This cherry pick is non-trivial due to merge conflicts (the file was moved and changed quite a bit since 0.71), so we will skip it for the current patch. @jpdriver, do you think you could prepare a PR to the 0.71-stable branch?

**Reply by ken0nek** [reply]
@fortmarek Thank you for handling the release!

> so we will skip it for the current patch

Understood 👍 

This is the patch I'm using for my project.

Extract `extract_projects` and `exclude_i386_architecture_while_using_hermes` from main.

```diff
diff --git a/scripts/cocoapods/utils.rb b/scripts/cocoapods/utils.rb
--- a/scripts/cocoapods/utils.rb
+++ b/scripts/cocoapods/utils.rb
@@ -53,22 +53,36 @@ class ReactNativePodsUtils
         end
     end

-    def self.exclude_i386_architecture_while_using_hermes(installer)
-        projects = installer.aggregate_targets
+    def self.extract_projects(installer)
+        return installer.aggregate_targets
             .map{ |t| t.user_project }
             .uniq{ |p| p.path }
             .push(installer.pods_project)
+    end

+    def self.exclude_i386_architecture_while_using_hermes(installer)
+        is_using_hermes = self.has_pod(installer, 'hermes-engine')

-        # Hermes does not support `i386` architecture
-        excluded_archs_default = ReactNativePodsUtils.has_pod(installer, 'hermes-engine') ? "i386" : ""
+        if is_using_hermes
+            key = "EXCLUDED_ARCHS[sdk=iphonesimulator*]"

-        projects.each do |project|
-            project.build_configurations.each do |config|
-                config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] = excluded_archs_default
-            end
+            projects = self.extract_projects(installer)

-            project.save()
+            projects.each do |project|
+                project.build_configurations.each do |config|
+                    current_setting = config.build_settings[key] || ""
+
+                    excluded_archs_includes_I386 = current_setting.include?("i386")
+
+                    if !excluded_archs_includes_I386
+                        # Hermes does not support `i386` architecture
+                        config.build_settings[key] = "#{current_setting} i386".strip
+                    end
+                end
+
+                project.save()
+            end
         end
     end
```

----

> do you think you could prepare a PR to the 0.71-stable branch?

Really appreciated. But I can prepare one if needed 🤝

**Reply by fortmarek** [reply]
@ken0nek if you can prepare a PR for 0.71, that will help us with adding it sooner 🙂 🙌

**Reply by ken0nek** [reply]
@fortmarek Created one https://github.com/facebook/react-native/pull/39763 👍

**Comment by cipolleschi** [comment]
Fixes for Xcode 15

https://github.com/facebook/react-native/commit/21763e85e39e17a19a1cf7a9026ef74517464749
https://github.com/facebook/react-native/commit/0dbd621c598e3ba7a203ec41bb70ce395ad1d62c
https://github.com/facebook/react-native/commit/8a5b2d673502037731ee6bc40fc64cdd22139011

**Reply by fortmarek** [reply]
Picked ✅

**Reply by fuekiin** [reply]
hi guys, is there an ETA for 0.71.14, since Xcode 15 is live today? Thanks!

**Reply by fortmarek** [reply]
We plan to release 0.71.14 next week.

**Comment by Yashtoddle** [comment]
Hi, 
the Hermes is causing the issue for the large json files and that issued is open from the past 2 year , and now that issue is resolved it would be great if you add this update in the coming patch . 
https://github.com/facebook/hermes/commit/3213794cba3cb21a901da4a36a7396a5889b5481

**Reply by fortmarek** [reply]
This won't make it to 0.71.14, we will look into it for the next patch.

**Reply by Yashtoddle** [reply]
what is the estimated date for the next patch , as I need this in my project urgently .

**Reply by lunaleaps** [reply]
@Yashtoddle There are merge conflicts getting this into 0.71. Can you make a PR against this branch https://github.com/facebook/hermes/tree/rn/0.71-stable and get the original author of https://github.com/facebook/hermes/commit/3213794cba3cb21a901da4a36a7396a5889b5481 to approve? Otherwise, the release crew is not equipped to support this pick

**Comment by cipolleschi** [comment]
Pick for Xcode 15:
- https://github.com/facebook/react-native/pull/39710

**Reply by fortmarek** [reply]
Backported via https://github.com/facebook/react-native/pull/39754

**Comment by cipolleschi** [comment]
Cherry-pick: https://github.com/facebook/react-native/commit/ce39931bc2b02f13cbc5751ba4d4a6dbc07bc91a

They released ActiveSupport 7.1.0 which breaks Cocoapods. 😮‍💨
This should fix the issue for all our users as they should use Bundler.

**Reply by fortmarek** [reply]
Picked ✅

**Comment by frankcalise** [comment]
Hello! Would it be possible to get https://github.com/facebook/react-native/pull/38256 picked up in 71?
